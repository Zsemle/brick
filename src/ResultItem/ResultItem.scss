@import '../style/layout.scss';

.result-item {
  border: 1px solid black;
  margin-bottom: $space-small;
}
