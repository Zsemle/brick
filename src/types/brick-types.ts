export type ExperienceCategory = 'animals'| 'sports' | 'brain games'

interface IPrice {
    amount: number
    currency: string
}

export class Price {
    amount: number
    currency: string

    constructor ({ amount, currency }:IPrice) {
      this.amount = amount
      this.currency = currency
    }
}

interface IExperience {
    name: string
    description: string
    category: ExperienceCategory
    price: Price
}

export class Experience {
    name: string
    description: string
    category: ExperienceCategory
    price: Price

    constructor ({ name, description, category, price }:IExperience) {
      this.name = name
      this.description = description
      this.category = category
      this.price = price
    }
}
