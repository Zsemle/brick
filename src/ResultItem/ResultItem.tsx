import React, { FunctionComponent } from 'react'
import { Experience } from '../types/brick-types'

const T:any = {
  name: 'Name:',
  description: 'Description:',
  category: 'Category:',
  price: 'Price:'
}

const ResultItem:FunctionComponent<Experience> = ({
  name,
  description,
  category,
  price
}:Experience):JSX.Element => {
  return (
    <li>
      <p>{T.name}{name}</p>
      <p>{T.description}{description}</p>
      <p>{T.category}{category}</p>
      <p>{T.price}{price.currency}{price.amount}</p>
    </li>
  )
}

export default ResultItem
