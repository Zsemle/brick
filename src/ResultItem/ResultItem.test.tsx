import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import ResultItem from './ResultItem'
import { Experience } from '../types/brick-types'

describe('the result item component', () => {
  it('renders the component', () => {
    const props:Experience = {
      name: 'test experience',
      description: 'description of the experience',
      category: 'brain games',
      price: {
        amount: 234,
        currency: 'USD'
      }
    }

    const wrapper:ShallowWrapper = shallow(<ResultItem {...props}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
