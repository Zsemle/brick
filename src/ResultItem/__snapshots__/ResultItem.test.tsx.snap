// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the result item component renders the component 1`] = `
<li>
  <p>
    Name:
    test experience
  </p>
  <p>
    Description:
    description of the experience
  </p>
  <p>
    Category:
    brain games
  </p>
  <p>
    Price:
    USD
    234
  </p>
</li>
`;
